tracer/build/_build/Build.Steps.cs

```diff
@@ -226,12 +246,39 @@ bool RequiresThoroughTesting()
             })
             .ToList();
 
+        // Check if any of the integration areas have changed
+        foreach (var file in gitChangedFiles)
+        {
+            foreach (var area in integrationAreaChanges.Keys.ToArray())
+            {
+                if (integrationAreaPaths[area].Any(file.Contains))
+                {
+                    integrationAreaChanges[area] = true;
+                    Logger.Information($"Integration area {area} has changed due to {file} - full test suite will be run for that area");
+                }
+            }
+        }
+
+        var changedAreas = integrationAreaChanges
+            .Where(kvp => kvp.Value)
+            .Select(kvp => kvp.Key)
+            .ToList();
+
+        // Set the environment variable with changed areas (even if we end up doing a full test)
+        // This allows tests to be selective even during a full test run if needed
+        // Note though that this can only be reached on PRs
+        if (changedAreas.Count != 0)
```

Comment:
I feel like we may as well declare a variable for it, since we reuse it afterwards and `hasIntegrationChanges` and `snapshotChangeCount` are also declared.

```suggestion
        bool areasChanged = changedAreas.Count != 0;
        if (areasChanged)
```

I don't think it's a big deal especially, since it will only be run in PRs. I just find it more readable 🙂